src/connectors/numeric-menu/connectNumericMenu.ts

```diff
@@ -4,146 +4,157 @@ import {
- *       {start: 10, name: 'more than 10'},
- *     ],
- *   })
- * );
- */
-export default function connectNumericMenu(renderFn, unmountFn = noop) {
+type Option = {
+  /**
+   * Name of the option
+   */
+  label: string;
+  /**
+   * Higher bound of the option (<=)
+   */
+  start: number;
+  /**
+   * Lower bound of the option (>=)
+   */
+  end: number;
+};
+
+type Item = {
+  /**
+   *  Name of the option.
+   */
+  label: string;
+  /**
+   * URL encoded of the bounds object with the form `{start, end}`.
+   * This value can be used verbatim in the webpage and can be read by `refine`
+   * directly. If you want to inspect the value, you can do:
+   * `JSON.parse(window.decodeURI(value))` to get the object.
+   */
+  value: string;
+  /**
+   * True if the value is selected
+   */
+  isRefined: boolean;
+};
+
+type ParamTransformItems = (items: Item[]) => Item[];
+
+export type NumericMenuConnectorParams = {
+  /**
+   * Name of the attribute for filtering
+   */
+  attribute: string;
+  /**
+   * List of all the items
+   */
+  items: Option[];
+  /**
+   * Function to transform the items passed to the templates
+   */
+  transformItems: ParamTransformItems;
+};
+
+type Refine = (facetValue: string) => void;
+
+export interface NumericMenuRendererOptions<TNumericMenuWidgetParams>
+  extends RendererOptions<TNumericMenuWidgetParams> {
+  /**
+   * The list of available choices
+   */
+  items: Item[];
+  /**
+   * Creates URLs for the next state, the string is the name of the selected option
+   */
+  createURL: (value: Item['value']) => string;
+  /**
+   * `true` if the last search contains no result
+   */
+  hasNoResults: boolean;
+  /**
+   * Sets the selected value and trigger a new search
+   */
+  refine: Refine;
+}
+
+export type NumericMenuRenderer<TNumericMenuWidgetParams> = Renderer<
+  NumericMenuRendererOptions<
+    NumericMenuConnectorParams & TNumericMenuWidgetParams
+  >
+>;
+
+export type NumericMenuWidgetFactory<TNumericMenuWidgetParams> = WidgetFactory<
+  NumericMenuConnectorParams & TNumericMenuWidgetParams
+>;
+
+type NumericMenuConnector = <TNumericMenuWidgetParams>(
+  render: NumericMenuRenderer<TNumericMenuWidgetParams>,
+  unmount?: () => void
+) => NumericMenuWidgetFactory<TNumericMenuWidgetParams>;
+
+const connectNumericMenu: NumericMenuConnector = (
+  renderFn,
+  unmountFn = noop
+) => {
   checkRendering(renderFn, withUsage());
 
-  return (widgetParams = {}) => {
-    const { attribute, items, transformItems = x => x } = widgetParams;
+  return widgetParams => {
+    const {
+      attribute = '',
+      items = [],
+      transformItems = (x => x) as ParamTransformItems,
+    } = widgetParams || {};
 
-    if (!attribute) {
+    if (attribute === '') {
       throw new Error(withUsage('The `attribute` option is required.'));
     }
 
-    if (!items) {
+    if (!items || items.length === 0) {
       throw new Error(
         withUsage('The `items` option expects an array of objects.')
```

Comment:
The error message here is not longer compatible with the condition:
when `items = []`, it says: "we expect an array"

Reply:
I don't really think this is a problem